frontend-react/src/components/ManagePublicKey/ManagePublicKeyChooseSender.tsx

```diff
@@ -0,0 +1,59 @@
+import React, { useState } from "react";
+import { Form, Dropdown, Button, FormGroup } from "@trussworks/react-uswds";
+
+import { UseOrganizationSenders } from "../../hooks/UseOrganizationSenders";
+import Spinner from "../Spinner";
+
+type Props = {
+    onSenderSelect: any; //What type should this be???
+};
+
+export function ManagePublicKeyChooseSender({ onSenderSelect }: Props) {
+    const { isLoading, senders } = UseOrganizationSenders();
+    const [selectedSender, setSelectedSender] = useState(
+        senders?.length === 1 ? senders[0].name : ""
+    );
+
+    if (senders?.length === 1) {
```

Comment:
This means that `onSenderSelect` will fire every time this component renders, which can happen A LOT based on state updates.  At the very least, it should be moved into an effect:

```ts
useEffect(() => {
    if (senders.length === 1) {
        onSenderSelect(selectedSender);
    }
}, [senders.length]); // just using senders.length as a primitive comparison value, not very intelligent
```

However, this should probably be done earlier on so this step can be skipped over entirely in the first place 🤔

Reply:
I originally had the logic in ManagePublicKey prior to breaking this out. Should I move it back there and pass the senders to ManagePublicKeyChooseSender?

Reply:
Actually this component/logic will only get called initially if org has one sender and then onSenderSelect is a vallback to the parent to set the selectedSender.
```
if (senders.length === 1) {
        onSenderSelect(selectedSender);
    }
```